### PR TITLE
Ignore empty translations

### PR DESCRIPTION
**Pull request by hishamco**
Address #34

### CODE REVIEW

**Comment by hishamco** [comment]
This is tricky because the header is different, I will struggle to make this happen

**Comment by hishamco** [comment]
Almost done!! nothing but there's a weird behavior, some of the translations are dropped here, but it's not in my laptop 🤔

**Comment by hishamco** [comment]
Woohoo .. everything works as expected

:shipit:

File: pack.ps1 [hunk 1]
```diff
@@ -112,18 +112,56 @@ function buildNuGetMetaPackageSpec($pkgName)
...
 }
 
+function isEmptyTranslation($culture)
+{
+    $currentTranslationsFolder = [IO.Path]::Combine($localizationFolderName, $culture)
+    $defaultTranslations = Get-ChildItem $localizationFolderName -File   
+    $currentTranslations = Get-ChildItem $currentTranslationsFolder -File
+    $nonTranslatedContents = 0
+    $contentDiff = 14;
+    $minNonTranslatedContents = 61
```
Comment (hishamco) [inline pack.ps1]: I can try to eliminate some of the crazy numbers, if we can ensure that all `.po` files are well formatted. Have a look to `OrchardCore.Users.Core.po` in `bg-BG` culture for instance

Comment (hishamco) [reply pack.ps1]: /cc @agriffard

**Comment by hishamco** [comment]
@sebastienros can we consider this before 1.0.0 release while it's working. We can update this later to avoid the random that you asked for, believe it or not I set those numbers after a lot of exterminates

At least we can void publish an empty translation packages to NuGet

**Comment by sebastienros** [comment]
What about having a hard coded list of the language we should generate instead? We would update this list if someone provides enough entries for this language. Could be in the script directly.

**Comment by hishamco** [comment]
We could but this can't guarantee that the list is updated unless someone remind as or it could be a bot if new entries for an empty culture has been added

Again I see the current script works well for our solving the issue at least

**Comment by sebastienros** [comment]
Does this "magic number" logic still work?

Another suggestion would be to change this to generate all packages, and we'll publish all of them, but use a list of cultures to define which packages are added to the meta package.

That will allow us to actually define when a culture is added by default. No more magic numbers, and we can decide what cultures are complete enough before showing up.

This can be a list hard-coded in the script, or a json file in the repository which might contain more metadata later on.

**Comment by sebastienros** [comment]
for instance:

```json
{
  "cultures": [
  {
    "name": "ar",
    "display-name": "Arabic",
    "generate-package": true,
    "meta-package": true
  },
  {
    "name": "fr",
    "display-name": "French",
    "generate-package": true,
    "meta-package": true
  }
  ]
}
```

**Comment by hishamco** [comment]
> Does this "magic number" logic still work?

I will check the appveyor logs

> Another suggestion would be to change this to generate all packages, and we'll publish all of them, but use a list of cultures to define which packages are added to the meta package.

Let me think about it after I check the logs

**Comment by hishamco** [comment]
As you said earlier we can avoid the magic strings at all and go with your suggestion

> and we'll publish all of them

I don't see any value to publish an empty translations, but may be this will encourage who download the an empty package to go to Crowdin

**Comment by hishamco** [comment]
@agriffard @sebastienros let us decide which culture shouldn't be included in the meta package, so I can update the `cultures.json`

We will keep this updated everytime we want to include or exclude a culture

**Comment by hishamco** [comment]
My suggestion to exclude the following list at least:

- Croatian

- Danish

- Finnish

- Hungarian

-Indonesian

- Macedonian

- Norwegian

- Serbian Cyrillic

- Serbian Latin

- Slovak

- Slovenian

**Comment by sebastienros** [comment]
Here is a list of culture and their score. First number is how many strings are missing from the Setup screen (very important to me, for tr.oc.net), and second number is total strings

arabic 0 74%
chinese simplified 0 92%
chinese traditional 0 21%
czech 0 66%
french 0 100%
german0 72%
greek (1) 0 89%
italian 0 99%
japanese (3) 22%
persian 1 81%
polish 3 37%
russian 3 65%
spanish 1 70%
swedish 44%
turkish 8 50%
vietnamese 0 88%

I suggest to add all the ones up as they either have a good setup coverage, or a good overall one. Also I think we can quickly fill up the blanks strings for Setup by just pinging some authors or asking  the community.

The next list is for the remaining language with poor Setup coverage. For these anything that has a string should be generated, but not added to the meta package

bulgarian 5%
croatian 0%
danish 0%
dutch 2%
finnish 0%
hebrew 7%
hungarian 0%
indonesian 0%
korean 0%
lithuanian 7%
macedonian 0%
norwegian 0%
portuguese 3%
portuguese brazilian 1%
serbian cyrillic 0%
serbian latin 0%
slovak 0%
slovenian 0%
ukranian 11%

And finally, all cultures should be added to the file.

**Comment by sebastienros** [comment]
I updated the file

**Comment by agriffard** [comment]
Do we also change the supportedCultures accordingly in the setup module?

```
private string[] _supportedCultures = new string[] {
            "ar", "cs", "da", "de", "el", "en", "es", "fa", "fi", "fr", "he", "hr", "hu", "id", "it", "ja", "ko", "lt", "mk", "nl", "pl", "pt", "ru", "sk", "sl", "sr-cyrl-rs", "sr-latn-rs", "sv", "tr", "uk", "vi", "zh-CN", "zh-TW"
        };
```

**Comment by sebastienros** [comment]
Right, only the one with the translations that are in the meta package

**Comment by hishamco** [comment]
> I updated the file

Thanks, I don't have internet connection last two days, @agriffard do you update the supported cultures list or shall I do it

**Pull request merged**